### ISSUE TITLE
Connections from neurons to nodes dont work. (v0.3.0)

### ISSUE THREAD

**Issue by xchoo** [nengo/nengo-loihi#105]
Observed behaviour that making a connection from neurons to a node doesn't properly process the spikes to produce an output. This behaviour works in v.0.2.0 but not in v.0.3.0.

The code:
```
import numpy as np
import matplotlib.pyplot as plt
import nengo
import nengo_loihi
from nengo.utils.matplotlib import rasterplot

with nengo.Network() as model:
    stim = nengo.Node(lambda t: np.sin(t * np.pi * 5))
    ens = nengo.Ensemble(2, dimensions=1, encoders=[[1], [-1]])
    nengo.Connection(stim, ens)
    
    output = nengo.Node(size_in=1)
    nengo.Connection(ens.neurons, output, transform=[[1, 1]])
    
    p1 = nengo.Probe(stim)
    p2 = nengo.Probe(ens.neurons)
    p3 = nengo.Probe(output)
```

**Comment by xchoo** [comment]
With `nengo_loihi v0.3.0`:
```
with nengo_loihi.Simulator(model, target='sim', precompute=True, seed=1231) as sim:
    sim.run(1)
plt.figure(figsize=(10, 8))
plt.subplot(3, 1, 1)
plt.plot(sim.trange(), sim.data[p1])
plt.title('Stimulus')
plt.subplot(3, 1, 2)
rasterplot(sim.trange(), sim.data[p2])
plt.title('Spikes')
plt.subplot(3, 1, 3)
plt.plot(sim.trange(), sim.data[p3])
plt.title('Output')
```
![download1](https://user-images.githubusercontent.com/1971583/46260214-86e30000-c4d2-11e8-9c03-81c1d5fd2759.png)

**Comment by xchoo** [comment]
With `nengo`:
```
with nengo.Simulator(model, seed=1231) as sim:
    sim.run(1)
plt.figure(figsize=(10, 8))
plt.subplot(3, 1, 1)
plt.plot(sim.trange(), sim.data[p1])
plt.title('Stimulus')
plt.subplot(3, 1, 2)
rasterplot(sim.trange(), sim.data[p2])
plt.title('Spikes')
plt.subplot(3, 1, 3)
plt.plot(sim.trange(), sim.data[p3])
plt.title('Output')
plt.show()
```

![download2](https://user-images.githubusercontent.com/1971583/46260222-9eba8400-c4d2-11e8-9435-ded4fe78170c.png)

**Comment by xchoo** [comment]
@pblouw informs me that this may be related or fixed with #103.

**Comment by tbekolay** [comment]
I'm pretty sure #103 won't fix this as the issue here is with a `.neurons` as `conn.post` while #103 deals with a `.neurons` as `conn.pre`. That said, both are issues that make it clear that the splitter needs heavy refactoring.

**Comment by drasmuss** [comment]
It looks like reverting #85 in #103 fixed this as well.  At least, the output is not a flat line anymore.  @xchoo could you confirm that the behaviour is what you would expect on the lastest `master`?

**Comment by xchoo** [comment]
Yup! I can confirm that on the latest `master` (#23163ad) the output is not flat any more, and looks like what I am expecting.

**Comment by drasmuss** [comment]
Sweet, closing this out then.  The fix will be in the next release (0.4).
